### DISCUSSION TITLE
To remote or not remote functions

### DISCUSSION THREAD

**Discussion by dennyglee** [unitycatalog/unitycatalog#364] [Ideas]
There is a good discussion [Slack](https://unitycatalog.slack.com/archives/C076YREKX8W/p1723448162628029?thread_ts=1723017584.499039&cid=C076YREKX8W) concerning functions and remote functions.  Let's move some of this discussion here so we can better track this.

**Comment by rickyma** [comment]
Thanks @dennyglee. 

My original question is: 
1. What is the difference between "Remote Functions" and "Functions"?
2. For the OSS version of remote functions, will they eventually be implemented as described in this document: https://docs.databricks.com/en/udf/unity-catalog.html

**Reply by dennyglee** [reply]
Thanks for asking @rickyma 

For your first question between "remote functions" and "functions"
* _Remote functions_ is a proposal on hosting tools for GenAI; we are still working on the proposal and will share it in a future milestone as based on feedback, we'd like to focus on AI assets like models and functions.
* _Functions_ is in reference to user-defined functions in SQL and Python so that it can be registered and then used within the context of UC.  

Concerning Unity Catalog itself, a good **starting point** for _functions_ is as described in the linked [UC UDFs](https://docs.databricks.com/en/udf/unity-catalog.html).  We've already started the process as you can see the [UC docs > functions](https://docs.unitycatalog.io/usage/functions/) and over the next few milestones, we will work on adding more ML and Spark integrations.
